File: mitza.jni4net/java/mitza/jni4net/wrapper/Iterable.java
package mitza.jni4net.wrapper;

import java.util.Iterator;

/**
 * @author Mihai Cadariu
 */
public class Iterable<E> implements java.lang.Iterable<E> {

	private system.collections.IEnumerable iEnumerable;

	public Iterable(system.collections.IEnumerable iEnumerable) {
		this.iEnumerable = iEnumerable;
	}

	@Override
	public Iterator<E> iterator() {
		return new IteratorBase<E>(iEnumerable.GetEnumerator());
	}

	@Override
	public String toString() {
		return String.format("%s<%s>", getClass().getName(), iEnumerable);
	}

	public system.collections.IEnumerable getIEnumerable() {
		return iEnumerable;
	}
}

File: branches/0-8-generics/mitza.jni4net/c#/Mitza/Jni4Net/Wrapper/Iterable.cs
using System.Collections.Generic;
using java.lang;

namespace Mitza.Jni4Net.Wrapper {

    public class Iterable<T> : IEnumerable<T> where T : java.lang.Object {

        private Iterable iterable;
        public Iterable WrappedIterable {
            get { return iterable; }
        }

        public Iterable(Iterable iterable) {
            this.iterable = iterable;
        }

        public IEnumerator<T> GetEnumerator() {
            return new Enumerator<T>(iterable);
        }

        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() {
            return new Enumerator<T>(iterable);
        }
    }
}
